src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -1404,8 +1409,13 @@ private void AwaitExpression(bool topLevel)
             }
         }
 
-        private void SetupDirectives()
+        private void SetupDirectives(IEnumerable<DirectiveDescriptor> directiveDescriptors)
         {
+            foreach (var directiveDescriptor in directiveDescriptors)
+            {
+                MapDirectives(() => HandleDirective(directiveDescriptor), directiveDescriptor.Name);
+            }
+
             MapDirectives(TagHelperPrefixDirective, SyntaxConstants.CSharp.TagHelperPrefixKeyword);
```

Comment:
These will be able to go away once they're implemented ontop of the directive descriptors. Will file an issue.